docs/cookbook/adding-ajax-load-more-button-into-pagination.md

```diff
@@ -0,0 +1,145 @@
+# Adding Ajax Load More Button into Pagination
+
+In this cookbook, we will add a paginated brand list including ajax "load more" button to a product list page. After finishing the guide, you will know how to use multiple paginations on one page.  
+
+*Note: After this change you will have paginated also `/brands-list/` route.*
+
+## Implementation of Brand Pagination
+
+First we need to extend `Shopsys\FrameworkBundle\Model\Product\Brand\BrandRepository` by creating `BrandRepository.php` in `/src/Shopsys/ShopBundle/Model/Product/Brand` and we add `getPaginationResult` method.
+```php
+/**
+ * {@inheritDoc}
+ */
+public function getPaginationResult(
+    $page,
+    $limit
+) {
+    $queryBuilder = $this->getBrandRepository()->createQueryBuilder('b');
+    $queryBuilder->orderBy('b.name', 'asc');
+
+    /** @var \Shopsys\FrameworkBundle\Component\Paginator\QueryPaginator $queryPaginator */
+    $queryPaginator = new QueryPaginator($queryBuilder);
+
+    return $queryPaginator->getResult($page, $limit);
+}
+```
+
+Then we will extend `BrandFacade` and add `getPaginatedResult` method.
+```php
+/**
+ * {@inheritDoc}
+ */
+public function getPaginatedResult(
+    $page,
+    $limit
+) {
+    $paginationResult = $this->brandRepository->getPaginationResult(
+        $page,
+        $limit
+    );
+
+    return new PaginationResult(
+        $paginationResult->getPage(),
+        $paginationResult->getPageSize(),
+        $paginationResult->getTotalCount(),
+        $paginationResult->getResults()
+    );
+}
+```
+
+After we have extended `BrandRepository` and `BrandFacade` we will create aliases for them in `services.yml`.
```

Comment:
```suggestion
After we have extended `BrandRepository` and `BrandFacade` we need to set them to be used instead of the framework classes in our application. This is done via configuration in `services.yml`.
```